pkg/policy/policies.go

```diff
@@ -213,6 +268,489 @@ func (ps *Policies) FilterableInUserlandMap() map[*Policy]int {
+			fEqs.utsEqualities[uts] = eq
+		}
+		for uts := range utsEqualities.Equal {
+			eq := fEqs.utsEqualities[uts]
+			// Equal == 1, so set n bitmap bit
+			utils.SetBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.utsEqualities[uts] = eq
+		}
+
+		// CommFilters equalities
+		commEqualities := p.CommFilter.Equalities()
+		for comm := range commEqualities.NotEqual {
+			eq := fEqs.commEqualities[comm]
+			// NotEqual == 0, so clear n bitmap bit
+			utils.ClearBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.commEqualities[comm] = eq
+		}
+		for comm := range commEqualities.Equal {
+			eq := fEqs.commEqualities[comm]
+			// Equal == 1, so set n bitmap bit
+			utils.SetBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.commEqualities[comm] = eq
+		}
+
+		// BinaryFilters equalities
+		binaryEqualities := p.BinaryFilter.Equalities()
+		for binaryNS := range binaryEqualities.NotEqual {
+			eq := fEqs.binaryEqualities[binaryNS]
+			// NotEqual == 0, so clear n bitmap bit
+			utils.ClearBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.binaryEqualities[binaryNS] = eq
+		}
+		for binaryNS := range binaryEqualities.Equal {
+			eq := fEqs.binaryEqualities[binaryNS]
+			// Equal == 1, so set n bitmap bit
+			utils.SetBit(&eq.equalInPolicies, policyID)
+			utils.SetBit(&eq.equalitySetInPolicies, policyID)
+			fEqs.binaryEqualities[binaryNS] = eq
+		}
+	}
+
+	return nil
+}
+
+func (ps *Policies) createNewFilterMapsVersion(bpfModule *bpf.Module) error {
+	mapsNames := map[string]string{ // inner map name -> outer map name
+		UIDFilterMap:      UIDFilterMapVersion,
+		PIDFilterMap:      PIDFilterMapVersion,
+		MntNSFilterMap:    MntNSFilterMapVersion,
+		PidNSFilterMap:    PidNSFilterMapVersion,
+		UTSFilterMap:      UTSFilterMapVersion,
+		CommFilterMap:     CommFilterMapVersion,
+		CgroupIdFilterMap: CgroupIdFilterVersion,
+		BinaryFilterMap:   BinaryFilterMapVersion,
+	}
+
+	polsVersion := ps.Version()
+	for innerMapName, outerMapName := range mapsNames {
+		// use the map prototype to create a new map
+		eqFilterMap, err := bpfModule.GetMap(innerMapName)
+		if err != nil {
+			return errfmt.WrapError(err)
+		}
+
+		info, err := bpf.GetMapInfoByFD(eqFilterMap.FileDescriptor())
+		if err != nil {
+			return errfmt.WrapError(err)
+		}
+
+		btfFD, err := bpf.GetBTFFDByID(info.BTFID)
+		if err != nil {
+			return errfmt.WrapError(err)
+		}
+
+		opts := &bpf.BPFMapCreateOpts{
+			BTFFD:                 uint32(btfFD),
+			BTFKeyTypeID:          info.BTFKeyTypeID,
+			BTFValueTypeID:        info.BTFValueTypeID,
+			BTFVmlinuxValueTypeID: info.BTFVmlinuxValueTypeID,
+			MapFlags:              info.MapFlags,
+			MapExtra:              info.MapExtra,
+			MapIfIndex:            info.IfIndex,
+		}
+		newInnerMap, err := bpf.CreateMap(
```

Comment:
Do we also delete the map at some point? I don't think I saw it in the code

Reply:
No, currently we lack that. It should be tackled through the version expiration.